scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala

```diff
@@ -298,7 +298,12 @@ object node {
   // shape of this data should probably change, currently we leave it for backward compatibility
   object FragmentInputDefinition {
 
-    @JsonCodec case class FragmentParameter(name: String, typ: FragmentClazzRef)
+    @JsonCodec case class FragmentParameter(
+        name: String,
+        typ: FragmentClazzRef,
+        validationExpression: Option[Expression] = None,
+        validationFailedMessage: Option[String] = None,
```

Comment:
failed message should be available only when expression is defined

Reply:
Wrapped it up into another case class